src/Catty/ViewController/Continue&New/MaintainObject/MaintainScript/BrickCategoryOverviewController.swift

```diff
@@ -0,0 +1,130 @@
+/**
+ *  Copyright (C) 2010-2020 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+import UIKit
+
+@objc class BrickCategoryOverviewController: UIViewController, UICollectionViewDelegate, UICollectionViewDataSource {
+
+    var brickCategoryOverviewCollectionView: UICollectionView!
+    var categegoriesBricks = [BrickCategory]()
+    var scvc: ScriptCollectionViewController?
+
+    // MARK: Init
+    @objc override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        categegoriesBricks = CatrobatSetup.self.registeredBrickCategories()
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+    }
+
+    @objc(initWithNibName:bundle:scvc:) init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?, scvcP: ScriptCollectionViewController) {
```

Comment:
Since this class is always initialized with `nil` for the first two parameters, meaning that the `nibName` and `bundle` parameters are unused, I would recommend to make a new, required initializer which only takes the `scriptCollectionViewController` as a parameter, since that controller also required that attribute to be set. By doing so, you can change that attribute to non-optional (you can forward `nil` to the super initializer for `nibName` and `bundle`).